src/orion/algo/space.py

```diff
@@ -275,6 +299,8 @@ def sample(self, n_samples=1, seed=None):
 
         return samples
 
+    def cast_to_dimension_type(self, value):
+        return numpy.float64(value)
```

Comment:
This is a issue with the tests. Since we use `numpy` to generate the values inside the sample function, the tests were not passing using the `float` (which was initially the type used here.)
I've implemented a function `underlying_type` which currently returns `numpy.float64` but could be use to change the type at runtime and be framework agnostic.